### ISSUE TITLE
test_get_raw_tables_view fails on Python 2.7.2

### ISSUE THREAD

**Issue by ctb** [ged-lab/khmer#868]
```
  File "/Users/t/dev/khmer/tests/test_counting_hash.py", line 117, in test_get_raw_tables_view
    memv = memoryview(tab)
TypeError: cannot make memory view because object does not have the buffer interface
```
c.f. #671

**Comment by kdmurray91** [comment]
Oh balls. Will investigate.

I think the issue may be old/new style buffers in CPython???

I was examining how to efficiently pass these to Cython as memory views and hit upon a similar issue (to do with read only buffers).

**Comment by kdmurray91** [comment]
Although that's rather intriguing, as `type(tab) is buffer`. And you'd hope that a buffer exposed the buffer interface :grinning:.

**Comment by luizirber** [comment]
This also breaks Python 2.6, but I guess we're not aiming for 2.6 anymore.
``` python
Traceback (most recent call last):
  File "khmer/.tox/py26/lib/python2.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "khmer/tests/test_counting_hash.py", line 117, in test_get_raw_tables_view
    memv = memoryview(tab)
NameError: global name 'memoryview' is not defined
```

**Comment by kdmurray91** [comment]
Mmmh, not good. I'll add a fix for @luizirber's observation.

**Comment by kdmurray91** [comment]
I haven't been able to find any python bugs/changes that indicate this is a python problem, but that may be my poor google skills.

**Comment by kdmurray91** [comment]
@luizirber any chance you could post your tox.ini? or even check it in, if that's ok.

**Comment by luizirber** [comment]
I didn't make a PR because it needs more polishing, for now I'm using something that works but is also wrong (the ideal would be moving parts from the Makefile here, and then calling tox on the Makefile)

``` ini
[tox]
envlist=py26,py27

[testenv]
commands=
    make install-dependencies
    make clean
    make test
```

**Comment by ctb** [comment]
Note, further investigation suggests that it is a problem in Python 2.7.2 but not Python 2.7.6.

**Comment by kdmurray91** [comment]
Interesingly, while testing on another machine (supercomputer using ICC), I got this:

```
khmer/_khmermodule.cc: In function 'PyObject* hash_get_raw_tables(khmer_KCountingHash_Object*, PyObject*)':
khmer/_khmermodule.cc:691: error: reference to 'Byte' is ambiguous
third-party/zlib/zconf.h:368: error: candidates are: typedef unsigned char Byte
lib/khmer.hh:83: error:                 typedef unsigned char khmer::Byte
khmer/_khmermodule.cc:691: error: reference to 'Byte' is ambiguous
third-party/zlib/zconf.h:368: error: candidates are: typedef unsigned char Byte
lib/khmer.hh:83: error:                 typedef unsigned char khmer::Byte
khmer/_khmermodule.cc:691: error: 'table_ptrs' was not declared in this scope
error: command 'gcc' failed with exit status 1
```

I.e, zlib.h has a similar typedef for Byte.

Using `khmer::Byte` in `khmer/_khmermodule.cc`  fixed that, so I'll add that to #869.

**Comment by kdmurray91** [comment]
#869  fixed this, AFAICT
